src/qtt/algorithms/functions.py

```diff
@@ -473,14 +480,17 @@ def estimate_parameters_damped_sine_wave(x_data, y_data, exponent=2):
     decay_factor = (mean_left + laplace_factor) / (mean_right + laplace_factor)
 
     duration = x_data[-1] - x_data[0]
-    sample_rate = x_data.size / duration
+    sample_rate = (x_data.size - 1) / duration
     ramseyfreq = estimate_dominant_frequency(y_data, sample_rate=sample_rate)
+
     if A == 0:
         angle = 0
     else:
         n_start = max(min((y_data[0] - B) / A, 1), -1)
-        angle = -np.arcsin(n_start)
-    t2s = 2 * duration / decay_factor
+        angle_first_datapoint = -np.arcsin(n_start)
+        angle = angle_first_datapoint + 2 * np.pi * ramseyfreq * x_data[0]
+        angle = np.mod(np.pi + angle, 2 * np.pi) - np.pi
+    t2s = 2 * duration / (decay_factor)
```

Comment:
the braces around decay factor are not needed

Reply:
I would like to review the math in your pull request if you want but the reference in the function heading doesn't give me enough information to do that. Maybe describing in docstring more what happens may make this easier (I like the extensive descriptions in numpy that really help deciding if the function is accurate to use for the job or not)